vernac/vernacentries.ml

```diff
@@ -1066,10 +1066,9 @@ let vernac_end_segment ({v=id} as lid) =
 (* Libraries *)
 
 let warn_require_in_section =
-  let name = "require-in-section" in
-  let category = "deprecated" in
-  CWarnings.create ~name ~category
-    (fun () -> strbrk "Use of “Require” inside a section is deprecated.")
+  CWarnings.create ~name:"require-in-section" ~category:"fragile"
+    (fun () -> strbrk "Use of “Require” inside a section is fragile." ++ spc() ++
+               strbrk "Committing files which use this functionality is not recommended.")
```

Comment:
I'd suggest; "using this functionality in production-ready Coq code is not recommended"

Reply:
Still the warning, while better now, it is not very informative; how could we improve this without adding a very long story?

Reply:
I would say "in finished proof scripts" to highlight the fact that this can be useful in the middle of a work.